### PR TITLE
chore: added activity report downloads

### PR DESCRIPTION
**Pull request by madhuravius**
added activity reports:

image

empty:

image

download works:

image

image

### CODE REVIEW

File: src/deploy/activity-report/index.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    yield* next();
+
+    if (!ctx.json.ok) {
+      return;
+    }
+
+    const url = ctx.json.data;
+
+    const link = document.createElement("a");
```
Comment (madhuravius) [inline src/deploy/activity-report/index.ts:109]: not ideal, not sure what other alternative there is without using blobs / garbage js loader code (which always sucks imo)

File: src/ui/pages/environment-detail-activity-reports.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  fetchEnvActivityReports,
+  selectActivityReportsByEnvId,
+} from "@app/deploy";
+import { AppState } from "@app/types";
+import { useSelector } from "react-redux";
+import { useParams } from "react-router";
+import { useQuery } from "saga-query/react";
+
+const ActivityReportsHeader = () => (
+  <div className="bg-white py-8 px-8 shadow border border-black-100 rounded-lg">
```
Comment (madhuravius) [inline src/ui/pages/environment-detail-activity-reports.tsx:16]: needs some styling tlc

File: src/ui/shared/environment/environment-activity-report-list.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+const ActivityReportListRow = ({
+  report,
+}: {
+  report: DeployActivityReport;
+}) => {
+  const dispatch = useDispatch();
+  return (
+    <tr className="group hover:bg-gray-50" key={`${report.id}`}>
+      <Td>{prettyEnglishDateWithTime(report.createdAt)}</Td>
```
Comment (madhuravius) [inline src/ui/shared/environment/environment-activity-report-list.tsx:18]: date format isn't quite what's in the mocks as we don't have a util for this without it showing up wrong

**Comment by joeclayallday** [comment]
Made my design tweaks
• Removed the big header section and replaced it with one line to the right of the table count "Weekly CSV downloads that list all environment operations."
• Right aligned the actions column and made buttons primary color styling

Screen Shot 2023-08-25 at 7 55 45 AM

**Review by neurosnap — APPROVED** [review]

**Pull request merged**